code/game/objects/structures/window.dm

```diff
@@ -180,50 +180,65 @@
 		return
 	..()
 
-/obj/structure/window/attackby(obj/item/I, mob/living/user, params)
+/obj/structure/window/tool_act(mob/living/user, obj/item/tool, tool_type, is_right_clicking)
 	if(!can_be_reached(user))
 		return TRUE //skip the afterattack
-
 	add_fingerprint(user)
+	return ..()
 
-	if(I.tool_behaviour == TOOL_WELDER)
-		if(atom_integrity < max_integrity)
-			if(!I.tool_start_check(user, amount = 0))
-				return
+/obj/structure/window/welder_act(mob/living/user, obj/item/tool)
+	if(atom_integrity < max_integrity)
+		if(!tool.tool_start_check(user, amount = 0))
+			return FALSE
 
-			to_chat(user, span_notice("You begin repairing [src]..."))
-			if(I.use_tool(src, user, 40, volume = 50))
-				atom_integrity = max_integrity
-				update_nearby_icons()
-				to_chat(user, span_notice("You repair [src]."))
-		else
-			to_chat(user, span_warning("[src] is already in good condition!"))
-		return
+		to_chat(user, span_notice("You begin repairing [src]..."))
+		if(tool.use_tool(src, user, 4 SECONDS, volume = 50))
+			atom_integrity = max_integrity
+			update_nearby_icons()
+			to_chat(user, span_notice("You repair [src]."))
+	else
+		to_chat(user, span_warning("[src] is already in good condition!"))
+	return TRUE
 
-	if(!(flags_1&NODECONSTRUCT_1) && !(reinf && state >= RWINDOW_FRAME_BOLTED))
-		if(I.tool_behaviour == TOOL_SCREWDRIVER)
-			to_chat(user, span_notice("You begin to [anchored ? "unscrew the window from":"screw the window to"] the floor..."))
-			if(I.use_tool(src, user, decon_speed, volume = 75, extra_checks = CALLBACK(src, .proc/check_anchored, anchored)))
-				set_anchored(!anchored)
-				to_chat(user, span_notice("You [anchored ? "fasten the window to":"unfasten the window from"] the floor."))
-			return
-		else if(I.tool_behaviour == TOOL_WRENCH && !anchored)
-			to_chat(user, span_notice("You begin to disassemble [src]..."))
-			if(I.use_tool(src, user, decon_speed, volume = 75, extra_checks = CALLBACK(src, .proc/check_state_and_anchored, state, anchored)))
-				var/obj/item/stack/sheet/G = new glass_type(user.loc, glass_amount)
-				if (!QDELETED(G))
-					G.add_fingerprint(user)
-				playsound(src, 'sound/items/deconstruct.ogg', 50, TRUE)
-				to_chat(user, span_notice("You successfully disassemble [src]."))
-				qdel(src)
-			return
-		else if(I.tool_behaviour == TOOL_CROWBAR && reinf && (state == WINDOW_OUT_OF_FRAME) && anchored)
-			to_chat(user, span_notice("You begin to lever the window into the frame..."))
-			if(I.use_tool(src, user, 100, volume = 75, extra_checks = CALLBACK(src, .proc/check_state_and_anchored, state, anchored)))
-				state = RWINDOW_SECURE
-				to_chat(user, span_notice("You pry the window into the frame."))
-			return
+/obj/structure/window/screwdriver_act(mob/living/user, obj/item/tool)
```

Comment:
note to self: check whether this condition is right

Reply:
oh this totally missed the spot I thought I was commenting LMAO 
it's the >= check on the window state